packages/ui-file-drop/src/FileDrop/styles.js

```diff
@@ -0,0 +1,132 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import generateComponentTheme from './theme'
+
+/**
+ * Generates the style object from the theme and provided additional information
+ * @param  {Object} theme The actual theme object.
+ * @param  {Object} themeOverride User provided overrides of the default theme mapping.
+ * @param  {Object} props the props of the component, the style is applied to
+ * @param  {Object} state the state of the component, the style is applied to
+ * @return {Object} The final style object, which will be used in the component
+ */
+const generateStyle = (theme, themeOverride, props, state) => {
+  // get the theme variables object for the component
+  const componentTheme = generateComponentTheme(theme, themeOverride)
+  const { height } = props
+  const {
+    functionallyDisabled,
+    visuallyDisabled,
+    dragRejected,
+    dragAccepted
+  } = state
+
+  const dragAcceptedStyle = dragAccepted
+    ? {
+        borderColor: componentTheme.acceptedColor
+      }
+    : {}
+
+  const dragRejectedStyle = dragRejected
+    ? {
+        borderColor: componentTheme.rejectedColor
+      }
+    : {}
+
+  const functionallyDisabledStyle = functionallyDisabled
+    ? {
+        cursor: 'not-allowed',
+        pointerEvents: 'none'
+      }
+    : {}
+
+  const visuallyDisabledStyle = visuallyDisabled
+    ? {
+        opacity: 0.5
+      }
+    : {}
+
+  // Force the content to fill the space set with the 'height' React prop
+  const heightStyle = height
+    ? {
+        height: '100%'
+      }
+    : {}
+
+  return {
+    fileDrop: {
+      label: 'fileDrop',
+      position: 'relative'
+    },
+    fileDropLabel: {
+      label: 'fileDrop__label',
+      display: 'block',
+      boxSizing: 'border-box',
+      position: 'relative',
+      ...heightStyle,
+      ...functionallyDisabledStyle,
+      ...visuallyDisabledStyle
+    },
+    fileDropInput: {
+      label: 'fileDrop__input',
+      width: '0.0625rem',
+      height: '0.0625rem',
+      margin: '-0.0625rem',
+      padding: 0,
+      position: 'absolute',
+      overflow: 'hidden',
+      clip: 'rect(0 0 0 0)',
+      border: 0
+    },
+    fileDropLabelContent: {
+      label: 'fileDrop__labelContent',
+      display: 'block',
+      boxSizing: 'border-box',
+      zIndex: 1,
+      textAlign: 'center',
+      borderRadius: componentTheme.borderRadius,
+      border: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`,
+      cursor: 'pointer',
+      '&:hover': {
+        // dragRejected should take precedence!
+        borderColor: dragRejected
+          ? componentTheme.rejectedColor
+          : componentTheme.hoverBorderColor
```

Comment:
This was accomplished a bit differently in the original, there they had another CSS class for rejected, so it took precedence over `:hover`

Reply:
Nice solution, but missing a little detail: as I can see in the original, the accepted state takes precedence too. It is not visible, because in the theme the hover and accepted color are set to the same blue color, but the component should handle the difference, so I suggest this:
```js
'&:hover': {
  ...(!dragAccepted && !dragRejected && { borderColor: componentTheme.hoverBorderColor })
},
```

Reply:
nice catch!